selfie.c

```diff
@@ -5512,12 +5514,12 @@ void emit_data_segment() {
   allocated_memory = 0;
 }
 
-uint64_t* create_elf_header(uint64_t binary_length) {
+uint64_t* create_elf_header(uint64_t binary_length, uint64_t code_length) {
   uint64_t* header;
 
   // store all numbers necessary to create a minimal and valid
   // ELF64 header including the program header
-  header = smalloc(ELF_HEADER_LEN);
+  header = touch(zalloc(ELF_HEADER_LEN), ELF_HEADER_LEN);
```

Comment:
That doesn't make sense. zalloc zeroes the allocated space. Touching is therefore subsumed by that.

Reply:
@ckirsch Actually, this (or something similar) is necessary, since at boot levels >0, zalloc falls back to malloc, which doesn't guarantee that the memory is mapped. If you remove this call to touch and run 'make all', the test fail for this reason. I have kept the call to touch and added a clarification comment, though I'm not sure if there's a better solution.

Reply:
Interesting! Let me suggest to clean this up a bit more. There are two code locations in which ELF headers are allocated. One in which zalloc is used, and one in which smalloc is used. Let's have instead just one short function called allocate_ELF_header which uses zalloc and touch.

Reply:
Done - If I'm not missing anything, all requested code changes are done